x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/ilm/TimeseriesLifecycleTypeTests.java

```diff
@@ -552,364 +552,6 @@ public void testGetOrderedActionsDelete() {
-        assertPreviousPhaseName("warm", null, new String[] {});
-
-        assertPreviousPhaseName("cold", "warm", new String[] { "hot", "warm", "cold", "delete" });
-        assertPreviousPhaseName("cold", "hot", new String[] { "hot", "cold", "delete" });
-        assertPreviousPhaseName("cold", "warm", new String[] { "warm", "cold", "delete" });
-        assertPreviousPhaseName("cold", null, new String[] { "cold", "delete" });
-        assertPreviousPhaseName("cold", "warm", new String[] { "hot", "warm", "delete" });
-        assertPreviousPhaseName("cold", "hot", new String[] { "hot", "delete" });
-        assertPreviousPhaseName("cold", "warm", new String[] { "warm", "delete" });
-        assertPreviousPhaseName("cold", null, new String[] { "delete" });
-        assertPreviousPhaseName("cold", "warm", new String[] { "hot", "warm" });
-        assertPreviousPhaseName("cold", "hot", new String[] { "hot" });
-        assertPreviousPhaseName("cold", "warm", new String[] { "warm" });
-        assertPreviousPhaseName("cold", null, new String[] {});
-
-        assertPreviousPhaseName("delete", "cold", new String[] { "hot", "warm", "cold", "delete" });
-        assertPreviousPhaseName("delete", "cold", new String[] { "warm", "cold", "delete" });
-        assertPreviousPhaseName("delete", "warm", new String[] { "hot", "warm", "delete" });
-        assertPreviousPhaseName("delete", "hot", new String[] { "hot", "delete" });
-        assertPreviousPhaseName("delete", "cold", new String[] { "cold", "delete" });
-        assertPreviousPhaseName("delete", null, new String[] { "delete" });
-        assertPreviousPhaseName("delete", "cold", new String[] { "hot", "warm", "cold" });
-        assertPreviousPhaseName("delete", "cold", new String[] { "warm", "cold" });
-        assertPreviousPhaseName("delete", "warm", new String[] { "hot", "warm" });
-        assertPreviousPhaseName("delete", "hot", new String[] { "hot" });
-        assertPreviousPhaseName("delete", "cold", new String[] { "cold" });
-        assertPreviousPhaseName("delete", null, new String[] {});
-
-        IllegalArgumentException exception = expectThrows(
-            IllegalArgumentException.class,
-            () -> TimeseriesLifecycleType.INSTANCE.getPreviousPhaseName("foo", Collections.emptyMap())
-        );
-        assertEquals("[foo] is not a valid phase for lifecycle type [" + TimeseriesLifecycleType.TYPE + "]", exception.getMessage());
-        exception = expectThrows(
-            IllegalArgumentException.class,
-            () -> TimeseriesLifecycleType.INSTANCE.getPreviousPhaseName(
-                "foo",
-                Collections.singletonMap("foo", new Phase("foo", TimeValue.ZERO, Collections.emptyMap()))
-            )
-        );
-        assertEquals("[foo] is not a valid phase for lifecycle type [" + TimeseriesLifecycleType.TYPE + "]", exception.getMessage());
-    }
-
-    public void testGetNextActionName() {
```

Comment:
We have dedicated methods to test the hot/warm etc actions e.g. https://github.com/elastic/elasticsearch/blob/main/x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/ilm/TimeseriesLifecycleTypeTests.java#L516 https://github.com/elastic/elasticsearch/blob/main/x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/ilm/TimeseriesLifecycleTypeTests.java#L526 etc